### PR TITLE
chore: Bump controller-runtime to v0.16.3

### PR DESCRIPTION
**Pull request by killianmuldoon**
Bump controller runtime to the latest release v0.16.3.

Release notes: v1.16.0 https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.16.0
Release notes: v1.16.3 https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.16.3

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 *.test
 testbin/
 cover.out
+.idea
```
Comment (adrianchiris) [inline .gitignore:6]: nit: newline

**Review by adrianchiris — APPROVED** [review]

**Review by adrianchiris — APPROVED** [review]
LGTM 🎸

**Review by cdesiniotis — APPROVED** [review]
lgtm. I have triggered the CI to run

**Comment by adrianchiris** [comment]
@killianmuldoon lets rebase and we can merge this one

**Review by adrianchiris — APPROVED** [review]

**Review by rollandf — APPROVED** [review]
LGTM

**Pull request merged**